### PR TITLE
Add waffles check to API

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

Add waffles check to API

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -33,6 +33,21 @@ jobs:
           ${{ runner.os }}-pip-
     - name: Install requirements
       run: pip install -r requirements_for_test.txt
+    - name: Setup upterm session
+      uses: lhotari/action-upterm@v1
+      with:
+        ## limits ssh access and adds the ssh public key for the user which triggered the workflow
+        limit-access-to-actor: true
+        ## limits ssh access and adds the ssh public keys of the listed GitHub users
+        limit-access-to-users: jzbahrai
```
Comment (andrewleith) [inline .github/workflows/test.yaml:42]: What does this do?  Should your username be in here?

Comment (jzbahrai) [reply .github/workflows/test.yaml]: ah i did it for debugging, it allows you to ssh into the box.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -33,6 +33,15 @@ jobs:
           ${{ runner.os }}-pip-
     - name: Install requirements
       run: pip install -r requirements_for_test.txt
+    - name: Run tests
```
Comment (andrewleith) [inline .github/workflows/test.yaml:36]: Is this related to waffles?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -36,12 +36,38 @@ jobs:
...
+      working-directory: /opt/hostedtoolcache/Python/${{ env.python_version }}/x64/lib/python3.9/
+      shell: bash
+      run: |
+        mkdir -p "${{ github.workspace }}/env/" && cp -fR site-packages "${{ github.workspace }}/env/"
+    - name: Install development .env file
+      working-directory: ${{ github.workspace }}
+      shell: bash
+      run: |
+        cp -f .env.example .env
+    - name: Checks for new endpoints against AWS WAF rules
```
Comment (andrewleith) [inline .github/workflows/test.yaml:63]: This seems really cool.  Think we could do a little sync on how this works?

Comment (sastels) [reply .github/workflows/test.yaml:63]: 💯 We should talk a bit about waffles at the next dev review or whatever meeting. It's really nice.

**Review by sastels — APPROVED** [review]
Awesome!
